main/main.cpp

```diff
@@ -1,445 +1,382 @@
-
-        case '(':
-            return from_elements({MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dash, MorseElement::Dot}, word_end);
-
-        case ')':
-            return from_elements({MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dash, MorseElement::Dot, MorseElement::Dash}, word_end);
-
-        case '&':
-            return from_elements({MorseElement::Dot, MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dot}, word_end);
-
-        case ':':
-            return from_elements({MorseElement::Dash, MorseElement::Dash, MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dot}, word_end);
-
-        case ';':
-            return from_elements({MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot}, word_end);
-
-        case '=':
-            return from_elements({MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dot, MorseElement::Dash}, word_end);
-
-        case '+':
-            return from_elements({MorseElement::Dot, MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot}, word_end);
-
-        case '-':
-            return from_elements({MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dot, MorseElement::Dot, MorseElement::Dash}, word_end);
-
-        case '_':
-            return from_elements({MorseElement::Dot, MorseElement::Dot, MorseElement::Dash, MorseElement::Dash, MorseElement::Dot, MorseElement::Dash}, word_end);
-
-        case '"':
-            return from_elements({MorseElement::Dot, MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot}, word_end);
-
-        case '$':
-            return from_elements({MorseElement::Dot, MorseElement::Dot, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot, MorseElement::Dot, MorseElement::Dash}, word_end);
-
-        case '@':
-            return from_elements({MorseElement::Dot, MorseElement::Dash, MorseElement::Dash, MorseElement::Dot, MorseElement::Dash, MorseElement::Dot}, word_end);
-
-        default:
-            return {{MorseElement::MarkSep}};  // PH error
-        }
-        // clang-format on
-    }
-};
+        // Use PID on the error and add a feed forward term which assumes constant acceleration motion (trapezoid motion
+        // profile).
+        return meters_per_second{
+            std::copysignf(kv, err) * std::sqrt(2 * Motor::max_acceleration * std::abs(err)) + pid.calculate_pid(err)
+        };
+    };
+    const auto limit_velocity = [&](float new_velocity, float last_velocity)
+    {
+        static constexpr auto max_accelaration =
+            Motor::max_acceleration
+            * duration_cast<std::chrono::duration<float, std::ratio<1>>>(pid_loop_period_us).count();
```

Comment:
IIRC `std::ratio<1>` is the default ratio.

```suggestion
            * duration_cast<std::chrono::duration<float>>(pid_loop_period_us).count();
```